lib/run_loop/core.rb

```diff
@@ -170,7 +164,7 @@ def self.run_with_options(options)
       uia_strategy = options[:uia_strategy]
       if uia_strategy == :host
         create_uia_pipe(repl_path)
-        RunLoop::HostCache.default.clear
+        RunLoop::HostCache.default.clear unless RunLoop::Environment.xtc?
```

Comment:
A different pull request maybe?  I didn't see this until the third pass through the code.

Was this causing problems on the XTC?